File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/procedures_ifreturn.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeProceduresIfreturn(statement: JSONObject, onError: (e: Exception) -> Unit) {
    try {
        Log.d("hello", "execute_procedures_ifreturn")
        val pointer = statement.getJSONObject("pointer")
        val condition = executeStatement(pointer["condition"].toString()) as? Boolean ?: false
        val value = executeStatement(pointer["value"].toString())
        if (condition) {
            functionReturnValue = value
            isFunctionReturn = true
        }
    } catch (e: Exception) {
        onError(e)
    }
}

File: piper/extensions/chrome/src/engine/statements/procedures_ifreturn.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let condition = await this.executeStatement(stringify(pointer.condition))
        let value = await this.executeStatement(stringify(pointer.value))

        if (condition) {
            this.functionReturnValue = value
            this.isFunctionReturn = true
        }
    } catch (e) {
        onError(e)
    }
}
